File: Assets/Scripts/Animation.js
//CURRENTLY NOT FUNCTIONAL

#pragma strict

public var FramesPerSecond : float;
public var FrameNumber : Texture[];
public var LoopFrames : boolean;

private var StopAnimation : boolean;
private var WaitingTime;
private var CurrentFrame : int;


function Start(){
	CurrentFrame = 0;
	WaitingTime = 1/FramesPerSecond;
	Animate();
}

function Update(){
	if(StopAnimation != true){
		Animate();
	}
}

function Animate(){
	StopAnimation = false;	
	if(CurrentFrame >= FrameNumber.Length){
		if(LoopFrames == true){
			CurrentFrame = 0;
		}
	}
	
	yield new WaitForSeconds(WaitingTime);
	renderer.material.mainTexture = FrameNumber[CurrentFrame];
	CurrentFrame++;
}
	

File: Assets/Scripts/Animation.cs
//For animations -> Convert to javascript. srsly

using UnityEngine;
using System.Collections;

public class Animation : MonoBehaviour {
	

	public float FramesPerSecond;
	private float WaitingTime;
	public Texture[] FrameNumber;
	public bool LoopFrames;
	
	private int CurrentFrame;
	
	
	void Start () {
		CurrentFrame = 0;
		WaitingTime = 1/FramesPerSecond;
		StartCoroutine(Animate ());
	}
	
	IEnumerator Animate(){
		bool StopAnimation = false;	
		if(CurrentFrame >= FrameNumber.Length){
			if(LoopFrames == true){
				CurrentFrame = 0;
			}
		
		}
		
		yield return new WaitForSeconds(WaitingTime);
		renderer.material.mainTexture = FrameNumber[CurrentFrame];
		CurrentFrame++;
		
		if(StopAnimation != true){
			StartCoroutine(Animate());
		}
		
	}
	
}
